python/pip.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -e
```

Comment:
General comment for this file: the file already complex, why not in python instead?

Reply:
See line 17.  I was halfway through rewriting one of these scripts in Python before I realized: I can't invoke a `py_binary` during `WORKSPACE`, I'd have to publish a PAR file.  I immediately `git stash`ed and rewrote my TODO instead. :(

Reply:
:( I actually do things without py_binary for simple python_script (I know that is a shame) using simply python <script.py>. Anyway up to you.

Reply:
I suppose that is true.  I want to be able to factor libraries and have a clean separation of unit tests etc.  I'll experiment with my stashed changes with this in mind.

Reply:
The only remaining shell script now does this (I copied it from you :D)